wgpu-hal/src/dx12/shader_compilation.rs

```diff
@@ -84,25 +84,30 @@ mod dxc {
     use std::path::PathBuf;
 
     pub(crate) struct DxcContainer {
-        pub compiler: hassle_rs::DxcCompiler,
-        pub library: hassle_rs::DxcLibrary,
+        compiler: hassle_rs::DxcCompiler,
+        library: hassle_rs::DxcLibrary,
         // Has to be held onto for the lifetime of the device otherwise shaders will fail to compile
         _dxc: hassle_rs::Dxc,
+        validator: hassle_rs::DxcValidator,
+        _dxil: hassle_rs::Dxil,
     }
 
     pub(crate) fn get_dxc_container(
         dxc_path: Option<PathBuf>,
         dxil_path: Option<PathBuf>,
     ) -> Result<Option<DxcContainer>, crate::DeviceError> {
         // Make sure that dxil.dll exists.
-        match hassle_rs::Dxil::new(dxil_path) {
-            Ok(_) => (),
+        let dxil = match hassle_rs::Dxil::new(dxil_path) {
+            Ok(dxil) => dxil,
             Err(e) => {
                 log::warn!("Failed to load dxil.dll. Defaulting to Fxc instead: {}", e);
                 return Ok(None);
             }
         };
 
+        // Needed for explicit validation.
+        let validator = dxil.create_validator()?;
```

Comment:
I guess we could if people are okay with Microsoft potentially yelling at them? People would still need to get and distribute `dxcompiler.dll` so I don't consider also requiring `dxil.dll` to be much of a problem. It would be one minor unblocker for future Linux support for wgpu's dx12 backend, but I'm not sure if that's ever going to be something you'd consider supporting?

Not supporting it also means the match in the `compile_dxc(...)` function doesn't get worse than it already is.

Reply:
Fair enough, I'm not too worried about Microsoft here as it's dxc generating the code, but at least for now it's fair to require both.